rust/regorule-library/src/userinfo/v1.rego

```diff
@@ -0,0 +1,23 @@
+package stackable.opa.userinfo.v1
+
+# Lookup by (human-readable) username
+userInfoByUsername(username) := http.send({
+  "method": "POST",
+  "url": "http://127.0.0.1:9476/user",
+  "body": {"username": username},
+  "headers": {"Content-Type": "application/json"},
+  "raise_error": true
+}).body
+
+# Lookup by stable user identifier
+userInfoById(id) := http.send({
+  "method": "POST",
+  "url": "http://127.0.0.1:9476/user",
+  "body": {"id": id},
+  "headers": {"Content-Type": "application/json"},
+  "raise_error": true
+}).body
+
+# Lookup by context
+currentUserInfoByUsername := userInfoByUsername(input.username)
+currentUserInfoById := userInfoById(input.id)
```

Comment:
I'd prefer to not ship this. The username is passed differently in our products!

E.g. hdfs has
```
        "callerUgi": {
            "shortUserName": "admin",
            "userName": "admin/test-hdfs-permissions.default.svc.cluster.local@CLUSTER.LOCAL",
        },
```
and Trino looks entirely different
```suggestion
```